IHP/Controller/Context.hs

```diff
@@ -137,5 +138,10 @@ instance HasField "frameworkConfig" ControllerContext FrameworkConfig where
     getField controllerContext = controllerContext.requestContext.frameworkConfig
     {-# INLINABLE getField #-}
 
+-- The following hack is bad, but allows us to override the logger using 'putContext'
+-- The alternative would be https://github.com/digitallyinduced/ihp/pull/1921 which is also not very nice
+--
+-- This design mistake should be fixed in IHP v2
```

Comment:
Out of curiosity, what's your idea for fixing on v2?

Reply:
No specific idea yet. We kind of want to make controller context less mutable (e.g. we could  change the type signature from `initContext` to the type signature of `mapContext` in the other branch. That way it's more functional, which would avoid this issue)